@here/harp-mapview/lib/geometry/TileGeometryManager.ts

```diff
@@ -80,36 +79,22 @@ export class TileGeometryManager {
      */
     constructor(protected mapView: MapView) {}
 
-    /**
-     * Initialize the {@link Tile} with the TileGeometryManager.
-     */
-    initTile(tile: Tile): void {
-        if (tile.dataSource.useGeometryLoader) {
-            tile.tileGeometryLoader = new TileGeometryLoader(tile, this.mapView.taskQueue);
-        }
-    }
-
     /**
      * Process the {@link Tile}s for rendering. May alter the content of the tile per frame.
      */
     updateTiles(tiles: Tile[]): void {
         let prio = 0;
         for (const tile of tiles) {
-            const geometryLoader = tile.tileGeometryLoader;
-            if (geometryLoader) {
-                //this assumes the tiles are ordered by priority, this is currently done in
-                // the visible tile set with 0 as the highest priority
-                geometryLoader.priority = prio++;
-            }
-
-            if (geometryLoader !== undefined) {
-                geometryLoader.update(
-                    this.enableFilterByKind ? this.enabledGeometryKinds : undefined,
-                    this.enableFilterByKind ? this.disabledGeometryKinds : undefined
-                );
-                if (this.m_tileUpdateCallback) {
-                    this.m_tileUpdateCallback(tile);
-                }
+            //this assumes the tiles are ordered by priority, this is currently done in
+            // the visible tile set with 0 as the highest priority
+            const tilePriority = prio++;
+            const updateDone = tile.updateGeometry(
+                tilePriority,
+                this.enableFilterByKind ? this.enabledGeometryKinds : undefined,
```

Comment:
Also, would it make sense to remove this logic to enable / disable geometry kinds at run time, and instead, when the kinds change, just reload the tiles.

That would simplify the logic quite a bit, instead of constantly calling `update`, it could just be one single call to await on the TileLoader to finish loading and then load the geometry.

Reply:
Makes sense. Could you bring it up in the chat and create a ticket if nobody sees any problem?

Reply:
would be great if we can get rid of it and remove the update function

Reply:
Yes, but let's do it in a separate ticket, otherwise this change is gonna blow up outside of the scope of this ticket.
I could do it after my current ticket.